### DISCUSSION TITLE
Can't update nn.dense.linear object inside nn.conv.nn_conv using .data

### DISCUSSION THREAD

**Discussion by swedenfish** [pyg-team/pytorch_geometric#6556] [Q&A] [answered]
Hi, I have a GNN model which has conv like this:
        
        nn = Sequential(Linear(6, 36), ReLU())
        self.conv1 = NNConv(1, 36, nn, aggr='mean')

And I am trying to update this model by manually updating its fields as below:
`model.conv1.nn[0].weight.data = main_model.conv1.nn[0].weight.data.clone()`
`model.conv1.nn[0].bias.data = main_model.conv1.nn[0].bias.data.clone()`
`model.conv1.bias.data = main_model.conv1.bias.data.clone()`
`model.conv1.lin.data = main_model.conv1.lin.data.clone()`

The first three works fine but the fourth one doesn't work, it is complained that **Linear** object doesn't have attributes **data**.
To be specific, Linear object here is the "**lin**" above, which is a [torch_geometric.nn.dense.linear](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/nn/dense/linear.html) object.
And it is an attribute declared inside [NNConv](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/nn/conv/nn_conv.html). 

Could I please have some helps on how I can update the model's conv1.lin? Thank you!!!

**Comment by wsad1** [comment, marked as answer]
This is expected. Instead try `lin.weight.data`.

**Reply by swedenfish** [reply]
Thanks for answering! But as before, I think I am trying to update the whole Linear object with this line. 
`model.conv1.lin.data = main_model.conv1.lin.data.clone()`
Is that equivalent to just updating the weight? What about other properties inside the Linear object?

**Reply by wsad1** [reply]
`model.conv1.lin` should have `weight` and `bias` you'll need to update both `weight.data` and `bias.data` in that case.

**Reply by swedenfish** [reply]
I think bias should be fine? Here is the source code for [NNConv](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/nn/conv/nn_conv.html#:~:text=if%20root_weight%3A%0A%20%20%20%20%20%20%20%20%20%20%20%20self.lin%20%3D%20Linear(in_channels%5B1%5D%2C%20out_channels%2C%20bias%3DFalse%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20weight_initializer%3D%27uniform%27)) 

`if root_weight:
            self.lin = Linear(in_channels[1], out_channels, bias=False,
                              weight_initializer='uniform')
`

In each NNConv object, it declares its inner Linear object with a default of not learning bias. 
I was trying to think about other parameters I would need to update in my codes.

**Reply by wsad1** [reply]
In that case there aren't any more parameters.

**Reply by swedenfish** [reply]
Thank you so much for your help!
